### PR TITLE
Fix metadata sync issue for cache listeners in multi-tab

### PR DESCRIPTION
**Pull request by milaGGL**
Issue: when creating a snapshot listener, it's `QueryTargetState` is simply set to "not-current" if it is listening to cache only. This is going to create out-of-sync metadata in persistence multi-tabs:

scenario: 
1. Create a snapshot listener in primary client, listen to server
2. It raised a snapshot from server, which is up-to-date (fromCache=false),  and store it into cache
3. Listen to same query with source==cache in secondary tab
4. Since the cached result is up-to-date, and the cache listener in the secondary tab is sharing the date with the server listener in the primary tab, the raised snapshot should be `current` as well (fromCache=false)

### CODE REVIEW

**Comment by DellaBitta** [comment]
Could you please provide a PR description. A detail of the change, a link to the bug, etc? Thanks!

File: packages/firestore/src/local/shared_client_state.ts [hunk 1]
```diff
@@ -676,9 +682,13 @@ export class WebStorageSharedClientState implements SharedClientState {
       }
     }
 
-    this.localClientState.addQueryTarget(targetId);
-    this.persistClientState();
+    // If the query is listening to cache only, the target ID should not be registered the  with
```
Comment (egilmorez) [inline packages/firestore/src/local/shared_client_state.ts:685]: A couple of issues with articles here:

"If the query is listening to cache only, the target ID should not be registered with the local Firestore client as an active watch target."

At least, I *think* the "an" is required.  WDYT Mark or others?

File: .changeset/warm-oranges-itch.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@firebase/firestore': patch
+---
+
+Fix metadata `fromCache` default to `true` issue when listen to cache in multi-tabs.
```
Comment (egilmorez) [inline .changeset/warm-oranges-itch.md:5]: Suggest phrasing more like:

"Fix an issue with metadata `fromCache` defaulting  to `true` when listening to cache in multi-tabs."

Should it actually be "multiple tabs?"

**Review by egilmorez — COMMENTED** [review]
A couple of things, thanks!

We should also get Mark Arndt's review on all things Firestore.

**Review by wu-hui — APPROVED** [review]

**Pull request merged**